server/src/main/java/org/opensearch/extensions/ExtensionActionListenerHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.extensions;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchException;
+
+/**
+ * Handles ActionListener requests from extensions
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionListenerHandler {
+
+    private static final Logger logger = LogManager.getLogger(ExtensionActionListener.class);
+    private ExtensionActionListener listener;
+
+    public ExtensionActionListenerHandler(ExtensionActionListener listener) {
+        this.listener = listener;
+    }
+
+    /**
+     * Handles a {@link ExtensionActionListenerOnFailureRequest}.
+     *
+     * @param request  The request to handle.
+     * @return A {@link AcknowledgedResponse} indicating success or failure.
+     */
+    public AcknowledgedResponse handleExtensionActionListenerOnFailureRequest(ExtensionActionListenerOnFailureRequest request) {
+        try {
+            listener.onFailure(new OpenSearchException(request.getFailureException()));
+            return new AcknowledgedResponse(true);
```

Comment:
This is not clear: the request certainly failed (we are in `handleExtensionActionListenerOnFailureRequest`), but we ` AcknowledgedResponse` as succeful?

Reply:
The AcknowledgedResponse being true just indicated that the request to act on the failure went through successfully.